accepted/2020/better-obsoletion/securestring-obsoletion.md

```diff
@@ -0,0 +1,221 @@
+
+## New overloads on existing APIs
+
+Some APIs - particularly those dealing with cryptographic keys - are ideal candidates for being enlightened about `SensitiveData<T>`. A non-exhaustive list is provided below.
+
+```cs
+namespace System.Security.Cryptography
+{
+    public sealed class AesGcm
+    {
+        /* existing ctors */
+        public AesGcm(byte[] key);
+        public AesGcm(ReadOnlySpan<byte> key);
+
+        /* new proposed ctor */
+        public AesGcm(SensitiveData<byte> key);
+    }
+
+    public class Rfc2898DeriveBytes
+    {
+        /* existing method */
+        public static byte[] Pbkdf2(byte[] password, byte[] salt, int iterations, HashAlgorithmName hashAlgorithm, int outputLength);
+
+        /* new proposed overload - takes and returns a SensitiveData parameter */
+        public static SensitiveData<byte> Pbkdf2(SensitiveData<byte> password, byte[] salt, int iterations, HashAlgorithmName hashAlgorithm, int outputLength);
+    }
+}
+```
+
+## New proposed SecureString-replacement APIs
+
+For the X.509 APIs, we should expose APIs which take the password as a `SensitiveData<char>` or a `SensitiveData<byte>` directly. These would serve as overloads of the existing `SecureString`-consuming APIs.
+
+```cs
+namespace System.Security.Cryptography.X509Certificates
+{
+    public class X509Certificate
+    {
+        public X509Certificate(byte[] rawData, SensitiveData<char> password, X509KeyStorageFlags keyStorageFlags);
+        public virtual void Import(byte[] rawData, SensitiveData<char> password, X509KeyStorageFlags keyStorageFlags);
+        /* ... */
+    }
+}
+```
+
+For `Process.Start` and `ProcessStartInfo` we should expose overloads which take a `NetworkCredential`. These overloads would only work on Windows.
+
+```cs
+namespace System.Diagnostics
+{
+    public class Process
+    {
+        public static Process Start(string fileName, NetworkCredential credential);
+    }
+
+    public class ProcessStartInfo
+    {
+        public NetworkCredential Credentials { get; set; }
+    }
+}
+```
+
+For `NetworkCredential` we should expose a property to accept a shrouded password.
+
+```cs
+namespace System.Net
+{
+    public class NetworkCredential
+    {
+        public SensitiveData<char> SensitivePassword { get; set; }
+    }
+}
+```
+
+Finally, there should be a conversion mechanism between `SecureString` and `SensitiveData<char>`. This should help transition developers who currently have SecureString instances to APIs which accept shrouded buffers, and vice versa (for interoping with legacy code).
```

Comment:
NetworkCredential should be in the list too - (1) use simple string to store password (on Linux at least), (2) conversion mechanism between NetworkCredential  and SecureString and SensitiveData<char> would be useful.

Reply:
It's already accounted for in the sentence immediately preceding this one.